libraries/src/Installer/InstallerHelper.php

```diff
@@ -74,58 +84,125 @@ public static function downloadPackage($url, $target = false)
         ini_set('user_agent', $version->getUserAgent('Installer'));
 
         // Load installer plugins, and allow URL and headers modification
-        $headers    = [];
+        $headers           = [];
+        $headers['accept'] = 'application/json, application/zip';
+
+        // Load installer plugins, and allow URL and headers modification
         $dispatcher = Factory::getApplication()->getDispatcher();
         PluginHelper::importPlugin('installer', null, true, $dispatcher);
         $event = new BeforePackageDownloadEvent('onInstallerBeforePackageDownload', [
-            'url'     => &$url, // @todo: Remove reference in Joomla 7, see BeforePackageDownloadEvent::__constructor()
-            'headers' => &$headers, // @todo: Remove reference in Joomla 7, see BeforePackageDownloadEvent::__constructor()
+            'url'     => &$url, // @todo: Remove reference in Joomla 6, see BeforePackageDownloadEvent::__constructor()
+            'headers' => &$headers, // @todo: Remove reference in Joomla 6, see BeforePackageDownloadEvent::__constructor()
         ]);
         $dispatcher->dispatch('onInstallerBeforePackageDownload', $event);
         $url     = $event->getArgument('url', $url);
         $headers = $event->getArgument('headers', $headers);
 
+        if (empty($url)) {
+            // Any logging and messaging of this are the responsibility of the event handlers.
+            return false;
+        }
+
         try {
-            $response = (new HttpFactory())->getHttp()->get($url, $headers);
+            $response = HttpFactory::getHttp()->get($url, $headers);
```

Comment:
```suggestion
            $response = (new HttpFactory())->getHttp()->get($url, $headers);
```
Please revert this unrelated change possibly caused by using an old file for working as this change reintroduced the deprecated static method call.